src/Workspaces/Core/Portable/Storage/SQLite/SQLitePersistentStorage_DocumentSerialization.cs

```diff
@@ -10,13 +10,13 @@ namespace Microsoft.CodeAnalysis.SQLite
     internal partial class SQLitePersistentStorage
     {
         public override Task<Checksum> ReadChecksumAsync(Document document, string name, CancellationToken cancellationToken)
-            => _documentAccessor.ReadChecksumAsync((document, name), cancellationToken);
+            => Task.FromResult(_documentAccessor.ReadChecksum((document, name), cancellationToken));
```

Comment:
Now that these are completely synchronous, maybe it's worth considering `ValueTask`?

Reply:
1. It's a public api.
2. I could easily expect many implementations of this api